sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1561,9 +1562,9 @@ object CodeGenerator extends Logging {
   private def logGeneratedCode(code: CodeAndComment): Unit = {
     val maxLines = SQLConf.get.loggingMaxLinesForCodegen
     if (Utils.isTesting) {
-      logError(s"\n${CodeFormatter.format(code, maxLines)}")
+      logError(log"\n${MDC(FORMATTED_CODE, CodeFormatter.format(code, maxLines))}")
```

Comment:
IMO this may not be a `LogKey` case.

Typically, `*_ID` stuffs are used as `LogKey` for log aggregating and filtering, e.g. we may want to collect the whole logs for a task. While for this case, the generated code is a large string block and mostly non-repeating, just leave it as the log content is sufficient.

Reply:
Yeah, I agree with you!

Reply:
Oh, so we only need to apply `logKey` that can literally be potentially used as "key", but in this case it would be a large non-repeating string, so wouldn't it be appropriate to apply `logKey`. Am I get it correctly?

Reply:
Sorry let me just ask some dumb questions for my understanding. Then how about this case?

https://github.com/apache/spark/blob/d487acd3008b6e8e9624295c833a6f87c1102ce9/sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala#L520-L522

Can we use `logKey` maybe something like `FIELD_NAME`??

Reply:
I think a more appropriate explanation is that if we need to query through `JSON` in `subsequent` queries, it can be used as a LogKey. 
According to the current log4j configuration, it will be output as a `field` to the `context`, eg:
![Uploading image.png…]()

Reply:
Thanks for your super fast review @pan3793 @panbingkun !

Then I think we have nothing to do for the SQL catalyst module on migrating `logError`, but let me wait for the @gengliangwang review as well before closing :-)